base/compiler/utilities.jl

```diff
@@ -96,6 +96,34 @@ function is_inlineable_constant(@nospecialize(x))
     return count_const_size(x) <= MAX_INLINE_CONST_SIZE
 end
 
+"""
+    is_inlineable(method::Method) -> Bool
+
+Check if `method` is eligible for inlining.
+"""
+function is_inlineable(method::Method)
+    isdefined(method, :source) || return false
+    return ccall(:jl_ir_flag_inlineable, Bool, (Any,), method.source)
```

Comment:
Perhaps make this `jl_ir_flag_inlineable` function return a UInt8 enum (or cost?) to DRY this pair of functions